daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Pretty.scala

```diff
@@ -179,6 +179,17 @@ private[lf] object Pretty {
             text(
               s"Found duplicated contract keys in submitted disclosed contracts for template $templateId"
             )
+          case DisclosurePreprocessing.NonExistentTemplate(templateId) =>
+            text(
+              s"Template $templateId does not exist"
```

Comment:
The error message here does not mention that this is about a disclosed contract.